### PR TITLE
ARG Query Generation API in 2023-09-01-preview

### PR DESCRIPTION
**Pull request by visingla-ms**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by visingla-ms** [comment]
Prettier check issues seem to be wrong, as locally running prettier checks, and it finds no issues.

**Comment by visingla-ms** [comment]
@raych1 For the Go DSK Breaking change, please see the comment from private repo PR, which has the same changes, but now being done in public repo: https://github.com/Azure/azure-rest-api-specs-pr/pull/14206#discussion_r1303688196.

File: specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2023-09-01-preview/resourcegraph.json [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+          "description": "Defines in which format query result returned.",
+          "type": "string",
+          "enum": [
+            "table",
+            "objectArray"
+          ],
+          "default": "objectArray",
+          "x-ms-enum": {
+            "name": "ResultFormat",
+            "modelAsString": false
```
Comment (TimLovellSmith) [inline specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2023-09-01-preview/resourcegraph.json:221]: I'm trying to remember if this needs to be modelAsString true, if you want it to be extensible in the future...

Comment (visingla-ms) [reply specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2023-09-01-preview/resourcegraph.json:221]: Hi Tim, this swagger is not updated, and this definition existed before.

Comment (visingla-ms) [reply specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2023-09-01-preview/resourcegraph.json:221]: Only resourcecopilot.json is newly added here.

Comment (TimLovellSmith) [reply specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2023-09-01-preview/resourcegraph.json:221]: Well you can't change from string without it being SDK breaking so probably simpler not to fix this ;(

**Comment by TimLovellSmith** [comment]
I don't think you really need to add suppression for this - can you please remove it? Its just a point-in-time approvable breaking change isn' tit? And we won't keep getting it in future revisions. But we would want to get warned, if you made more similar breaking changes.

```
  - suppress: XmsIdentifierValidation
    from: resourcecopilot.json
    where: $.definitions.Error.properties.details
    reason: Adding x-ms-identifiers to Error details array results in SDK breaking changes.
```

**Comment by TimLovellSmith** [comment]
> I don't think you really need to add suppression for this - can you please remove it? Its just a point-in-time approvable breaking change isn' tit? And we won't keep getting it in future revisions. But we would want to get warned, if you made more similar breaking changes.
> 
> ```
>   - suppress: XmsIdentifierValidation
>     from: resourcecopilot.json
>     where: $.definitions.Error.properties.details
>     reason: Adding x-ms-identifiers to Error details array results in SDK breaking changes.
> ```

Well never mind for now. Maybe we can remove it later.

**Comment by TimLovellSmith** [comment]
SDK breaking changes approval required before merge.

**Review by TimLovellSmith — APPROVED** [review]

**Comment by visingla-ms** [comment]
Thank you, @TimLovellSmith and @sjanamma for the approval.

**Comment by TimLovellSmith** [comment]
@rkmanda here is the PR that has the linter fatal errors again... any idea what is Nimma and whats broken?

2023-09-07T21:03:15.3478181Z {"pluginName":"spectral","extensionName":"@microsoft.azure/openapi-validator","level":"fatal","message":"spectralPluginFunc: Failed validating: 'file:///mnt/vss/_work/1/lint-c93b354fd9c14905bb574a8834c4d69b/specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/stable/2022-10-01/resourcegraph.json', error encountered: Error: Error running Nimma"}

**Pull request merged**